File: frontend/src/types/project.ts
export interface RepositorySummary {
  id: string;
  name: string;
  status: string;
  total_files: number;
  created_at: string;
}

export interface Project {
  id: string;
  name: string;
  description?: string;
  created_at: string;
  updated_at: string;
  repository_count: number;
  repositories: RepositorySummary[];
}

export interface ProjectCreateInput {
  name: string;
  description?: string;
}

export interface ProjectListResponse {
  total: number;
  projects: Project[];
}

File: backend/app/schemas/project.py
from datetime import datetime
from typing import Optional, List
from pydantic import BaseModel, Field, ConfigDict


class ProjectBase(BaseModel):
    name: str = Field(..., min_length=1, max_length=120, description="Project name")
    description: Optional[str] = Field(None, max_length=1000, description="Optional project description")


class ProjectCreate(ProjectBase):
    pass


class ProjectUpdate(BaseModel):
    name: Optional[str] = Field(None, min_length=1, max_length=120)
    description: Optional[str] = Field(None, max_length=1000)


class RepositorySummary(BaseModel):
    id: str
    name: str
    status: str
    total_files: int
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class ProjectResponse(ProjectBase):
    id: str
    created_at: datetime
    updated_at: datetime
    repository_count: int = 0
    repositories: List[RepositorySummary] = []

    model_config = ConfigDict(from_attributes=True)


class ProjectListResponse(BaseModel):
    total: int
    projects: List[ProjectResponse]
